src/Spryker/Client/RabbitMq/Model/Connection/ConnectionCreator/ConnectionCreator.php

```diff
@@ -0,0 +1,135 @@
+     */
+    protected $queueEstablishmentHelper;
+
+    /**
+     * @var \Spryker\Client\RabbitMq\Model\Connection\ConnectionInterface[]
+     */
+    protected $createdConnectionsMap;
+
+    /**
+     * @param \Spryker\Client\RabbitMq\Model\Helper\QueueEstablishmentHelperInterface $queueEstablishmentHelper
+     */
+    public function __construct(QueueEstablishmentHelperInterface $queueEstablishmentHelper)
+    {
+        $this->queueEstablishmentHelper = $queueEstablishmentHelper;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QueueConnectionTransfer $connectionConfig
+     *
+     * @return \Spryker\Client\RabbitMq\Model\Connection\ConnectionInterface
+     */
+    public function createConnectionByConfig(QueueConnectionTransfer $connectionConfig): ConnectionInterface
+    {
+        return $this->createOrGetConnection($connectionConfig);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QueueConnectionTransfer $connectionConfig
+     *
+     * @return \Spryker\Client\RabbitMq\Model\Connection\ConnectionInterface
+     */
+    protected function createOrGetConnection(QueueConnectionTransfer $connectionConfig): ConnectionInterface
+    {
+        if (isset($this->createdConnectionsMap[$connectionConfig->getName()])) {
+            return $this->createdConnectionsMap[$connectionConfig->getName()];
+        }
+
+        $connection = $this->createConnection($connectionConfig);
+        $this->createdConnectionsMap[$connectionConfig->getName()] = $connection;
+
+        return $connection;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QueueConnectionTransfer $queueConnectionConfig
+     *
+     * @return \Spryker\Client\RabbitMq\Model\Connection\ConnectionInterface
+     */
+    protected function createConnection(QueueConnectionTransfer $queueConnectionConfig): ConnectionInterface
+    {
+        return new Connection(
+            $this->createAmqpStreamConnection($queueConnectionConfig),
+            $this->queueEstablishmentHelper,
+            $queueConnectionConfig
+        );
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QueueConnectionTransfer $queueConnectionTransfer
+     *
+     * @return \PhpAmqpLib\Connection\AMQPStreamConnection
+     */
+    protected function createAmqpStreamConnection(QueueConnectionTransfer $queueConnectionTransfer): AMQPStreamConnection
+    {
+        $streamConnection = new AMQPStreamConnection(
+            $queueConnectionTransfer->getHost(),
+            $queueConnectionTransfer->getPort(),
+            $queueConnectionTransfer->getUsername(),
+            $queueConnectionTransfer->getPassword(),
+            $queueConnectionTransfer->getVirtualHost(),
+            $queueConnectionTransfer->getInsist() ?? false,
+            $queueConnectionTransfer->getLoginMethod() ?? 'AMQPLAIN',
+            $queueConnectionTransfer->getLoginResponse(),
+            $queueConnectionTransfer->getLocale() ?? 'en_US',
```

Comment:
mb, we can get default locale from somewhere? What if current store doesn't have `en_US` locale?

Reply:
Copypaste from `ConnectionFactory`. You are right, I'll fix this.